x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/restriction/WorkflowResolver.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.security.authz.restriction;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.Strings;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public final class WorkflowResolver {
+
+    private static final Logger logger = LogManager.getLogger(WorkflowResolver.class);
+
+    /**
+     * Allows access to Search Application query REST API.
+     */
+    public static final Workflow SEARCH_APPLICATION_QUERY_WORKFLOW = Workflow.builder()
```

Comment:
The `SEARCH_APPLICATION_QUERY_WORKFLOW` and `SEARCH_APPLICATION_ANALYTICS_WORKFLOW` are just a suggestion that might need to be changed depending on the result of an internal discussion.

Reply:
We synced internally and agreed that (for now) we only need a single workflow `search_application_query` that restricts access to `RestQuerySearchApplicationAction`. I've adjusted the workflows accordingly.